src/CabanaPD_BodyTerm.hpp

```diff
@@ -0,0 +1,52 @@
+/****************************************************************************
+ * Copyright (c) 2022-2023 by Oak Ridge National Laboratory                 *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of CabanaPD. CabanaPD is distributed under a           *
+ * BSD 3-clause license. For the licensing terms see the LICENSE file in    *
+ * the top-level directory.                                                 *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#ifndef BODYTERM_H
+#define BODYTERM_H
+
+#include <Kokkos_Core.hpp>
+
+#include <Cabana_Core.hpp>
+
+namespace CabanaPD
+{
+
+template <class UserFunctor>
+struct BodyTerm
+{
+    UserFunctor _user_functor;
+
+    BodyTerm( UserFunctor user )
+        : _user_functor( user )
+    {
+    }
+
+    // This purposely matches the boundary conditions.
```

Comment:
@streeve I do not understand this comment. In which sense this matches the boundary condition? Do we need this comment?

Reply:
Clarified

Reply:
@streeve I see in CabanaPD_Boundary.hpp the following (lines 157-168): 
    template <class ExecSpace, class ParticleType>
    void apply( ExecSpace, ParticleType& )
    {
        auto user = _user_functor;
        auto index_space = _index_space._view;
        Kokkos::RangePolicy<ExecSpace> policy( 0, index_space.size() );
        Kokkos::parallel_for(
            "CabanaPD::BC::apply", policy, KOKKOS_LAMBDA( const int b ) {
                auto pid = index_space( b );
                user( pid );
            } );
    }

Is this what you are referring to?

This is not identical but similar to:

   template <class ExecSpace, class ParticleType>
    void apply( ExecSpace, ParticleType& particles, const double time )
    {
        Kokkos::RangePolicy<ExecSpace> policy( 0, particles.n_local );
        auto user = _user_functor;
        Kokkos::parallel_for(
            "CabanaPD::BodyTerm::apply", policy,
            KOKKOS_LAMBDA( const int p ) { user( p, time ); } );
    }

For example, above we have "_index_space._view" instead of "particles.n_local". Why the difference?

Reply:
That's why I added "this function interface" matches - if the functions were identical we wouldn't need both.. this goes over all particles, BC goes over a subset.

Reply:
@streeve I see, thanks.